### PR TITLE
Topic recognition - Task 1 - 48241328

### PR DESCRIPTION
**Pull request by dbhasin4123**
Name: Deven Bhasin
ID: 48241328

Hi, I am submitting my implementation of the Improved UNet for the ISIC 2018 segmentation dataset (Task 1).

Trained models with this architecture was able to consistently achieve a dice similarity coefficient of greater or equal to 80% in training, validation, and testing.

The files to be added are:

dataset.py which defines process_images to process the images and create_ds for creating and splitting a dataset into training, validation, and test subsets.
module.py which defines the tensorflow implementation of the improved unet.
predict.py which defines functions for plotting images, displaying original, processed, and predicted images, and plotting performance metrics like loss and Dice Similarity Coefficient (DSC).
train.py which trains model with dice similarity coefficient and dice similarity coefficient loss.

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
This is an initial inspection, no action is required at this point.

- Wrong files path. All student constructed document should be packed in one document under the ‘recognition’ document.
- Algorithm solves the problem appropriately. Average Dice Coefficient in validating dataset: 0.8. However, predicted mask output is not binary one, so looks a little bit blur. Other parts are ok.
- Correctly using TensorFlow to construct the Improved U-Net models and functions. However, the training phase is implemented in the predict.py, using the training functions in the train.py. No usage of testing dataset for predicting and evaluating, though the raw dataset was split into train/validation/test datasets. Other parts are ok.
- Good coding design. Good handwriting code.
- Sufficient docstring and comment in the code.
- Easy difficulty problem, no higher difficulty marks.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present but needs better name than 'Results' (use ID/model?), also repo README overridden -2
No Usage & Demo & Visualisation & Data usage -1
Module present
Commenting
No Data leakage, but no proper testing set -1
Difficulty: Easy -10
## Commit Log
Meaningful commit messages sometimes only -1
Progressive commits used
## Documentation
ReadMe acceptable
Model/technical explanation needs more details -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, fix folder name and restore repo README -2
Request Description good

**Comment by shakes76** [comment]
Can recieve feedback marks +2 if the folder name and README is fixed.

**Comment by dbhasin4123** [comment]
Hi 
I fixed the folder name and README as per feedback.
I hope that 2 marks will be awarded.

Thanks

**Comment by wangzhaomxy** [comment]
Feedback attempt and feedback marks granted +2.
